### PR TITLE
Cleanup builder: remove container package dependency

### PR DESCRIPTION
**Pull request by tiborvass**
This PR cleans up the builder a little: remove the dependency on the container package, and removes the daemon dependency in the commit-specific code of the builder.

Ping @anusha-ragunathan @calavera @duglin

### CODE REVIEW

**Comment by LK4D4** [comment]
Pretty cool errors on CI: "Error must not be nil"

**Comment by anusha-ragunathan** [comment]
Several tests are failing. Will review again when all are green.

**Comment by cpuguy83** [comment]
It looks like ADD/COPY is not working, relevant logs on Windows: https://gist.github.com/cpuguy83/6d847f3a2dc2b2be5ca2
The ADD/COPY command does not outright fail, but files are not committed to new image.


```Dockerfile
FROM busybox
ADD foo /foo
RUN ls /foo
```

```
$ docker build -t test .
```

**Comment by tiborvass** [comment]
I put Mount() and Unmount() back for now until we come up with a better alternative.

**Comment by LK4D4** [comment]
Now windows fail seems unrelated.

File: builder/builder.go [hunk 1]
```diff
@@ -7,9 +7,10 @@ package builder
 import (
 	"io"
 	"os"
+	"time"
 
 	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon"
```
Comment (calavera) [inline builder/builder.go]: this is bringing the daemon dependency back, are we okay with it?

**Comment by anusha-ragunathan** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

File: builder/builder.go (hunk 1 above)
Comment (tiborvass) [inline builder/builder.go]: @calavera I'll remove it in another one it's just structs.

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**